testnet/stacks-node/src/mockamoto.rs

```diff
@@ -961,7 +957,28 @@ impl MockamotoNode {
         let config = self.chainstate.config();
         let chain_length = block.header.chain_length;
         let sortition_handle = self.sortdb.index_handle_at_tip();
-        let aggregate_public_key = self.self_signer.aggregate_public_key;
+        let aggregate_public_key = if chain_length <= 1 {
+            self.self_signer.aggregate_public_key
+        } else {
+            let block_sn = SortitionDB::get_block_snapshot_consensus(
+                sortition_handle.conn(),
+                &block.header.consensus_hash,
+            )?
+            .ok_or(ChainstateError::DBError(DBError::NotFoundError))?;
+            // TODO: https://github.com/stacks-network/stacks-core/issues/4109
```

Comment:
Is this meant to be addressed in this PR?

Reply:
No it is not meant to be addressed in this PR as far as I am aware. @kantai ?

Reply:
No, I don't think so.